src/main/java/uk/gov/hmcts/ccd/domain/model/common/CatalogueResponse.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.ccd.domain.model.common;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+@JsonDeserialize(builder = CatalogueResponse.Builder.class)
+public class CatalogueResponse<T> {
+
+    private final String code;
+    private final String message;
+    private final T details;
+
+    private CatalogueResponse(final String code, final String message, final T details) {
+        this.code = code;
+        this.message = message;
+        this.details = details;
+    }
```

Comment:
One more overload to take no details.

Reply:
Is it what is required default constructor ? 
public CatalogueResponse()
{
  super();
   this.code =null;
        this.message =null;
        this.details = null;
}

Reply:
I meant "private CatalogueResponse(final String code, final String message)"

Reply:
@Thor-tech-of-metal  - Kindly for your attention.. ^^^

Reply:
Perfect Mutlu .  I will do it :)